numalogic/registry/redis_registry.py

```diff
@@ -0,0 +1,247 @@
+import logging
+import time
+from typing import Optional, Any
+from collections.abc import Sequence
+
+from redis.client import Redis, Pipeline
+from redis.exceptions import RedisError
+
+from numalogic.registry import ArtifactManager, ArtifactData
+from numalogic.registry.serialize import loads, dumps
+from numalogic.tools.exceptions import ModelKeyNotFound
+from numalogic.tools.types import artifact_t
+
+_LOGGER = logging.getLogger()
+
+
+class RedisRegistry(ArtifactManager):
+    """
+    Model saving and loading using Redis Registry.
+    Args:
+        client: Take in the reids client already established/created
+        ttl: Total Time to Live for the key when saving in redis (dafault = 5000)
+
+    Examples
+    --------
+    >>> import redis
+    >>> from numalogic.models.autoencoder.variants import VanillaAE
+    >>> from numalogic.registry.redis_registry import RedisRegistry
+    >>> r = redis.StrictRedis(host='127.0.0.1', port=6379)
+    >>> cli = r.client()
+    >>> registry = RedisRegistry(client=cli)
+    >>> skeys = ['c', 'a']
+    >>> dkeys = ['d', 'a']
+    >>> model = VanillaAE(10)
+    >>> registry.save(skeys, dkeys, artifact=model, **{'lr': 0.01})
+    >>> registry.load(skeys, dkeys)
+    """
+
+    __slots__ = ("client", "ttl")
+    _HOST = None
+
+    def __new__(
+        cls,
+        client: Redis = None,
+        ttl: int = 604800,
+        *args,
+        **kwargs,
+    ):
+        instance = super().__new__(cls, *args, **kwargs)
+        if not cls._HOST:
+            cls._HOST = "None"
+        return instance
+
+    def __init__(
+        self,
+        client: Redis = None,
```

Comment:
We don't know what Redis client this would be so I would recommend creating a type var for this.